vortex-array/src/arrays/primitive/top_value.rs

```diff
@@ -0,0 +1,58 @@
+use std::hash::Hash;
+
+use rustc_hash::FxBuildHasher;
+use vortex_dtype::{NativePType, match_each_native_ptype};
+use vortex_error::{VortexExpect, VortexResult, vortex_bail};
+use vortex_mask::{AllOr, Mask};
+use vortex_scalar::PValue;
+
+use crate::Array;
+use crate::aliases::hash_map::HashMap;
+use crate::arrays::{NativeValue, PrimitiveArray};
+use crate::variants::PrimitiveArrayTrait;
+
+impl PrimitiveArray {
+    pub fn top_value(&self) -> VortexResult<(PValue, u32)> {
```

Comment:
Should write a docstring that documents behavior (error if self is empty or all null), or perhaps better, the return value should be `Option` or `VortexResult<Option<...>>`